src/main/histFileMod.F90

```diff
@@ -280,15 +288,16 @@ end subroutine copy_entry_interface
   ! tapes is assembled in the 'allhistfldlist' variable. Note that the first history tape is index 1 in
   ! the code but contains 'h0' in its output filenames (see set_hist_filename method).
   type history_tape
-     integer  :: nflds                         ! number of active fields on tape
-     integer  :: ntimes                        ! current number of time samples on tape
+     integer  :: nflds(maxsplitfiles)          ! number of active fields on file
+     integer  :: ntimes(maxsplitfiles)         ! current number of time samples on tape
      integer  :: mfilt                         ! maximum number of time samples per tape
      integer  :: nhtfrq                        ! number of time samples per tape
      integer  :: ncprec                        ! netcdf output precision
      logical  :: dov2xy                        ! true => do xy average for all fields
      logical  :: is_endhist                    ! true => current time step is end of history interval
      real(r8) :: begtime                       ! time at beginning of history averaging interval
-     type (history_entry) :: hlist(max_flds)   ! array of active history tape entries.
+     ! 13) DONE slevis: change hlist to (max_flds,maxsplitfiles)
+     type (history_entry) :: hlist(max_flds, maxsplitfiles)  ! array of active history tape entries.
                                                ! The ordering matches the allhistfldlist's.
```

Comment:
I think this comment is incorrect judging from this line found later in the same file. The n is a new counter of fields, while fld is the counter of all the fields:
`tape(t)%hlist(n,f)%field = allhistfldlist(fld)%field`

Reply:
Trying a clearer comment in the next commit and resolving this conversation.